omod/src/main/java/org/openmrs/module/fhir2/providers/AllergyIntoleranceFhirResourceProvider.java

```diff
@@ -93,4 +94,13 @@ public MethodOutcome createAllergy(@ResourceParam AllergyIntolerance allergy) {
 		return FhirServerUtils.buildCreate(fhirAllergyIntoleranceService.create(allergy));
 	}
 	
+	@Update
+	@SuppressWarnings("unused")
+	public MethodOutcome updateAllergy(@ResourceParam AllergyIntolerance allergy, @IdParam IdType id) {
+		if (id != null) {
+			allergy.setId(id.getIdPart());
+		}
```

Comment:
I don't love changing the resource id automatically. This allows easy errors (posting an updated resource to the wrong url) to potentially overwrite important information. I would just as soon not have this check here.